libmscore/tempotext.cpp

```diff
@@ -338,33 +338,51 @@ void TempoText::layout()
       adjustReadPos();
       }
 
+QString TempoText::duration2userName(const TDuration t)
+      {
+      QString dots;
+      switch (t.dots()) {
+            case 1: dots = tr("Dotted %1").arg(t.durationTypeUserName());
+                  break;
+            case 2: dots = tr("Double dotted %1").arg(t.durationTypeUserName());
+                  break;
+            case 3: dots = tr("Triple dotted %1").arg(t.durationTypeUserName());
+                  break;
+            default:
+                  dots = t.durationTypeUserName();
+                  break;
+            }
+      return dots;
+      }
+
 //---------------------------------------------------------
 //   accessibleInfo
 //---------------------------------------------------------
 
 QString TempoText::accessibleInfo() const
       {
-      TDuration t;
-      int len;
-      int x = findTempoDuration(plainText(), len, t);
-      if (x != -1) {
-            QString dots;
-
-            switch (t.dots()) {
-                  case 1: dots = tr("Dotted %1").arg(t.durationTypeUserName());
-                        break;
-                  case 2: dots = tr("Double dotted %1").arg(t.durationTypeUserName());
-                        break;
-                  case 3: dots = tr("Triple dotted %1").arg(t.durationTypeUserName());
-                        break;
-                  default:
-                        dots = t.durationTypeUserName();
-                        break;
+      TDuration t1;
+      TDuration t2;
+      int len1;
+      int len2;
+      QString text = plainText();
+      QString firstPart = text.split(" = ").first();
+      QString secondPart = text.split(" = ").back();
+      int x1 = findTempoDuration(firstPart, len1, t1);
+      int x2 = -1;
+      if (_relative)
+            x2 = findTempoDuration(secondPart, len2, t2);
+
+      if (x1 != -1) {
+            QString dots1;
+            QString dots2;
+            dots1 = duration2userName(t1);
+            if (x2 != -1) {
+                  dots2 = duration2userName(t2);
+                  return QString("%1: %2 %3 = %4 %5").arg(Element::accessibleInfo()).arg(dots1).arg(tr("note")).arg(dots2).arg(tr("note"));
                   }
-
-            QString bpm = plainText().split(" = ").back();
-
-            return QString("%1: %2 %3").arg(Element::accessibleInfo()).arg(dots).arg(tr("note = %1").arg(bpm));
+            else
+                  return QString("%1: %2 %3").arg(Element::accessibleInfo()).arg(dots1).arg(tr("note = %1").arg(secondPart));
```

Comment:
I don't really understand what this does - the whole translation stuff is
still kind of mysterious to me - but I'd just want to make sure this
doesn't negatively affect how a screenreader would read the text.

On Wed, May 18, 2016 at 10:59 AM Joachim Schmitz <notifications@github.com>
wrote:

> In libmscore/tempotext.cpp
> <https://github.com/musescore/MuseScore/pull/2607#discussion_r63741666>:
>
> >                    }
> > -
> > -            QString bpm = plainText().split(" = ").back();
> > -
> > -            return QString("%1: %2 %3").arg(Element::accessibleInfo()).arg(dots).arg(tr("note = %1").arg(bpm));
> > +            else
> > +                  return QString("%1: %2 %3").arg(Element::accessibleInfo()).arg(dots1).arg(tr("note = %1").arg(secondPart));
>
> we can even for 2.0.4, if need be
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/musescore/MuseScore/pull/2607/files/39c909355d0d17a14c089b496f217d4dfc895e0a..43c7250acbcdc7e95dbf2c8b8aef189f817d84ed#r63741666>
>